targets/curie_bsp/setup.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import sys
+import os
+
+# create soft link
+def build_soft_links(curie_path, jerry_path):
+  location_path = curie_path + '/wearable_device_sw/projects/curie_bsp_jerry'
+  if not os.path.exists(location_path):
+    os.makedirs(location_path)
+  os.chdir(location_path)
+  # arc
+  if not os.path.islink(location_path + '/arc'):
+    os.symlink(jerry_path + '/targets/curie_bsp/jerry_app/arc', 'arc')
+
+  # include
+  if not os.path.islink(location_path + '/include'):
+    os.symlink(jerry_path + '/targets/curie_bsp/jerry_app/include', 'include')
+  # quark
+  if not os.path.islink(location_path + '/quark'):
+    os.symlink(jerry_path + '/targets/curie_bsp/jerry_app/quark', 'quark')
+
+  # jerryscript
+  location_path += '/quark'
+  os.chdir(location_path)
+  if not os.path.islink(location_path + '/jerryscript'):
+    os.symlink(jerry_path, 'jerryscript')
+
+
+# create Kbuild.mk
+def breadth_first_travel(root_dir):
+  out_put = ''
+  lists = os.listdir(root_dir)
+  dirs = []
+  for line in lists:
+    full_path = os.path.join(root_dir, line)
+    if os.path.isdir(full_path):
+      dirs.append(line)
+    else:
+      c_file = line.endswith('.c')
+      if c_file:
+        npos = line.find('.c')
+        out_put += 'obj-y += ' + line[0:npos] + '.o\n'
+        continue
+      asm_file = line.endswith('.S')
+      if asm_file:
+        npos = line.find('.S')
+        out_put += 'obj-y += ' + line[0:npos] + '.o\n'
+  for line in dirs:
+    out_put += 'obj-y += ' + line + '/\n'
+  os.chdir(root_dir)
+  file_to_be_created = open('Kbuild.mk', 'w+')
+  file_to_be_created.write(out_put)
+  file_to_be_created.close()
+  for line in dirs:
+    breadth_first_travel(os.path.join(root_dir, line))
+
+
+
+# create Kbuild.mk in jerryscript/
+def create_kbuild_in_jerryscript(jerry_path):
+  breadth_first_travel(jerry_path + '/jerry-core')
+  breadth_first_travel(jerry_path + '/jerry-libm')
```

Comment:
If I understand correctly these two calls will create a few Kbuild.mk files inside `jerry-core` and `jerry-libm`. It would be preferable if we could keep those dirs clean, can't we create the `.mk` files into a different place?